ocaml-eglot.el

```diff
@@ -541,27 +466,10 @@ The universal prefix argument can be used to change the maximum number
 of results (LIMIT)."
   (interactive "sSearch query: \np")
   (ocaml-eglot--search-def-or-decl
-   #'ocaml-eglot-find-identifier-definition
-   query
-   limit))
-
-(defun ocaml-eglot-search-definition-in-current-window (query &optional limit)
-  "Search a definition using a QUERY (type or polarity) in the current window.
-The universal prefix argument can be used to change the maximum number
-of results (LIMIT)."
-  (interactive "sSearch query: \np")
-  (ocaml-eglot--search-def-or-decl
-   #'ocaml-eglot-find-identifier-definition-in-current-window
-   query
-   limit))
-
-(defun ocaml-eglot-search-definition-in-new-window (query &optional limit)
-  "Search a definition using a QUERY (type or polarity) in a new window.
-The universal prefix argument can be used to change the maximum number
-of results (LIMIT)."
-  (interactive "sSearch query: \np")
-  (ocaml-eglot--search-def-or-decl
-   #'ocaml-eglot-find-identifier-definition-in-new-window
+   (lambda (identifier)
+     (let ((ocaml-eglot-locate-preference 'ml)
+           (xref-prompt-for-identifier nil))
```

Comment:
You don't need to let-bind xref-prompt-for-identifier here as you aren't calling xref-find-definitions interactively